### PR TITLE
Hotfix/error on user profile navigation

### PR DESCRIPTION
**Pull request by maerzhase**
with the help of @louisholley superpower we figured it was an issue on the nextjs version :) 
further more fixed the warnings that `onClick` should be provided to children and not the `next/link` component
also updated the `@next/bundle-analyser` and the `next-eslint-config` to matching nextjs version

### CODE REVIEW

**Review by louisholley — COMMENTED** [review]
lgtm! will QA when the preview is deployed

File: src/utils/hookts.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const useClientEffect = (
   dependencies?: DependencyList
 ): void => {
   if (typeof window !== "undefined") {
-    useEffect(effect, dependencies)
+    useEffect(effect, dependencies) // eslint-disable-line
```
Comment (maerzhase) [inline src/utils/hookts.ts:63]: @GemN @louisholley this actually violates the hook rule that hooks can't be conditional. wdyt?

File: src/utils/hookts.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ export const useClientAsyncEffect = (
   dependencies?: any[]
 ): void => {
   if (typeof window !== "undefined") {
-    useAsyncEffect(effect, dependencies)
+    useAsyncEffect(effect, dependencies) // eslint-disable-line
```
Comment (maerzhase) [inline src/utils/hookts.ts:72]: @GemN @louisholley this actually violates the hook rule that hooks can't be conditional. wdyt?

File: src/utils/hookts.ts (hunk 1 above)
Comment (louisholley) [reply src/utils/hookts.ts:63]: i think it's fine in this case, `typeof window` will never go from undefined to undefined or vice versa so shouldn't give us any issues

File: src/utils/hookts.ts (hunk 2 above)
Comment (GemN) [reply src/utils/hookts.ts:72]: yeah that's right, you can't use a condition before a hook. But a useEffect is already client side (per react documentation) so this part should be removed imo

**Review by louisholley — APPROVED** [review]

**Pull request merged**